website/docs: 2024.2 release notes (#8468)
goauthentik/authentik: website/docs/releases/2024/v2024.2.md

:::::

## Highlights

-   **Remote Access Control** <span class="badge badge--primary">Enterprise</span> Access machines over RDP, SSH and VNC from authentik

-   **Audit log** <span class="badge badge--primary">Enterprise</span> See what fields were changed when objects are updated

-   **Session location and network binding** Increase security by preventing session theft

-   **Wizard to simplify creating applications and providers**

## Breaking changes

### Manual action is required

-   **Tenants have been renamed to brands**

    Tenants, which used to allow the changing of branding configuration and default flows and a couple other settings depending on which domain authentik was accessed from, have been renamed to _brands_. This is due to this release including full tenancy (see [here](#new-features))

    Existing _tenant_ objects will automatically be renamed to _brand_ objects. The API endpoints associated with _brands_ have also been renamed.

    Blueprints using `authentik_tenants.tenant` will need to be changed to use `authentik_brands.brand`.

>     See documentation for _brands_ [here](../../core/brands.md)

Reviewer: ```suggestion
    For more information, refer to the [documentation for _brands_](../../core/brands.md).
```

> 

Reviewer: The next two paragraphs are a bit confusing.. are they still part of the bullet item about Tenants, or separate? I think they are related, but for readability I'd combine those two into one paragraph. And... mve the link to the docs down to very last line of this bullet item.

Author: Also it says see above, where it should say see below where we talk about the system settings page

>     Also, **the event retention settings configured in brands (previously tenants, see above) has been removed and is now a system setting**, managed in the admin interface or via the API (see above).

Reviewer: ```suggestion
    Also, **the event retention settings configured in brands (previously tenants, see above) has been removed and is now a system setting**, managed in the Admin interface or via the API (see above).
```

>     **There is no built-in migration path for this change.** If you set something other than the default (`days=365`), you will need to update the setting in the admin interface.

Reviewer: I don't know what this means. (And that might be ok.)

-   **Helm chart breaking changes**

    The Helm Chart has a number of breaking changes. Find out more in the [chart release notes](https://github.com/goauthentik/helm/releases/tag/authentik-2024.2.0-rc1).

### Manual action may be required

-   **Required `offline_access` scope for Refresh tokens**

    The OAuth2 provider ships with a new default scope called `offline_access`, which must be requested by applications that need a refresh token. Previously, authentik would always issue a refresh token for the _Authorization code_ and _Device code_ OAuth grants.

>     Applications that require will need their configuration update to include the `offline_access` scope mapping.

Reviewer: Applications that require __what exactly?___ ....

Author: A refresh token, apparently I missed a word x)

-   **Database requirement changes**

    authentik now uses PostgreSQL schemas other than `public`.

    If you have a custom PostgreSQL deployment, please ensure that the authentik user is allowed to create schemas. Usually, if the authentik user is owner of the database, it already can.

-   **Redis and cache configuration options have been improved**

    Thank you @PKizzle for this contribution!

    Cache settings have been moved from the `redis` top-level config key to their own `cache` top-level config key.

    Settings have also been added to configure the Redis instance/database used for tasks and websockets seperately from cache. See [here](../../installation/configuration.mdx#redis-settings).

    Typically, _no changes to the configuration are required_.

> -   **Configuration options migrated to the admin interface**

Reviewer: ```suggestion
-   **Configuration options migrated to the Admin interface**
```

>     The following config options have been moved from the config file and can now be set using the admin interface (under **System** -> **Settings**) or the API:

Reviewer: ```suggestion
    The following config options have been moved from the config file and can now be set using the Admin interface (under **System** -> **Settings**) or the API:
```

    -   `AUTHENTIK_AVATARS`
    -   `AUTHENTIK_DEFAULT_USER_CHANGE_NAME`
    -   `AUTHENTIK_DEFAULT_USER_CHANGE_EMAIL`
    -   `AUTHENTIK_DEFAULT_USER_CHANGE_USERNAME`
    -   `AUTHENTIK_GDPR_COMPLIANCE`
    -   `AUTHENTIK_IMPERSONATION`
    -   `AUTHENTIK_FOOTER_LINKS`
    -   `AUTHENTIK_REPUTATION__EXPIRY`

    When upgrading to 2024.2, the currently configured options will be automatically migrated to the database, and can be removed from the `.env` or helm values file afterwards.

-   **Icons are now in a `public/` subfolder**

    If your media folder is `/media`, icons are now stored in `/media/public`. authentik will automatically migrate the icons upon upgrading.

    Note that even though that folder is named `public`, the files stored here are not automatically public. This is due to the naming of the default PostgreSQL schema.

-   **User sessions will be invalidated after this upgrade.**

>     As such, users will need to log back in. This will occur only once and does not require manual action from the user as they will be logged out automatically after the upgrade.

Reviewer: ```suggestion
    As such, users will need to log back in. Immediately after the upgrade completes, users are logged out automatically and are then prompted to log in again. This only occurs once.
```

    -   `authentik_outpost_ldap_requests_rejected` -> `authentik_outpost_ldap_requests_rejected_total`
    -   `authentik_outpost_proxy_requests` -> `authentik_outpost_proxy_request_duration_seconds`
    -   `authentik_outpost_proxy_upstream_time` -> `authentik_outpost_proxy_upstream_response_duration_seconds`
    -   `authentik_outpost_radius_requests` -> `authentik_outpost_radius_request_duration_seconds`
    -   `authentik_outpost_radius_requests_rejected` -> `authentik_outpost_radius_requests_rejected_total`
    -   `authentik_main_requests` -> `authentik_main_request_duration_seconds`

-   The shorthand parameter for `--stage`, `-s` for the `ak test_email` command has been changed to `-S`

## New features

-   **New provider: Remote Access Control** <span class="badge badge--primary">Enterprise</span>

    The Remote Access Control provider allows you to remotely connect to remote machines over RDP, SSH and VNC through authentik. As such, you can use the same policy engine and customization options that are possible with other providers using the same user and admin interface.

-   **Audit log** <span class="badge badge--primary">Enterprise</span>

>     authentik instances which have a valid enterprise license installed will log changes made to models including which fields were changed with previous and new values of the fields. The values are censored if they are sensitive (for example a password hash), however a hash of the changed value will still be logged.

Reviewer: ```suggestion
    authentik instances that have a valid enterprise license installed will log any changes made to models, including which fields were changed with previous and new values of the fields. The values are censored if they are sensitive (for example a password hash), however a hash of the changed value will still be logged.
```

-   **Session location and network binding**

>     Sessions for any users can now be bound to a specific geolocation (Continent, Country, City) or network (Autonomous System, subnet, IP address). If the session is accessed from a different location/network it was initially created from, the session will be terminated.

Reviewer: ```suggestion
    Sessions for any users can now be bound to a specific geolocation (Continent, Country, City) or network (Autonomous System, subnet, IP address). If the session is accessed from a location/network that is different than that from which it was initially created, the session will be terminated.
```

    Configuration steps are available [here](../../flow/stages/user_login/index.md#network-bindinggeoip-binding).

-   **S3 file storage**

>     Media files can now be stored on S3. Follow the [setup guide](../../installation/storage-s3.md) to get started.

Reviewer: ```suggestion
    Media files can now be stored on Amazon S3. Follow the [setup guide](../../installation/storage-s3.md) to get started.
```

Author: We are compatible with any S3 storage system, not just Amazon

-   **_Pretend user exists_ option for Identification stage**

>     Previously the identification stage would only continue if a user matching the user identifier exists. While this was the intended functionality, this release adds an option to continue to the next stage even if no matching user was found. "Pretend" users cannot authenticate nor receive emails, and don't exist in the database.

Reviewer: ```suggestion
    Previously the Identification stage would only continue if a user matching the user identifier exists. While this was the intended functionality, this release adds an option to continue to the next stage even if no matching user was found. "Pretend" users cannot authenticate nor receive emails, and don't exist in the database.
```

    **This feature is enabled by default.**

> -   **Tenancy** <span class="badge badge--primary">Enterprise</span>

Author: We said we'd put it all the way down because it's still alpha

    :::warning

>     This feature is in early preview. Use at your own risk.

Reviewer: Perhaps less scary wording, like "This feature is in early preview. We recommend backing up your instance before using." ??

Author: I like it scary. We don't have much experience with it, and until we do, I wouldn't recommend people use it.

    :::

>     It allows for authentik operators to manage several authentik installations without having to deploy additional instances.

Reviewer: ```suggestion
    With the release of 2024.2, an administrator or operator can now create multiple tenants. This means that an operator can manage several authentik installations without having to deploy additional instances.
    
    For more information about tenants, refer to the [documentation](../advanced/tenancy.md). Refer to the [brands documentation](../core/brands.md) for information about the updated process for configuring an instance's branding, colors, logos, etc.
```

### UX features

-   **Simplified workflow for creating applications and providers**

>     Applications and providers can now be created at the same time using the **Application Wizard**, found on the **Applications** -> **Applications** page of the admin interface, skipping the linking step between the two.

Reviewer: ```suggestion
    Applications and providers can now be created at the same time using the **Application Wizard**, found on the **Applications** -> **Applications** page of the Admin interface. The new wizard removes the previous requirement of first creating the provider, then the application, and then manually linking the two together.
```

-   **Ability to select more than 20 providers in an outpost**

    We have introduced a new way of selecting providers in the outpost configuration form, which allows for more than 20 providers to be selected for a single outpost.

    This dual-list multiselect prompt is only available for outposts for now, but we plan on extending it to other forms.

-   **Attribute preview per user**

    You can now preview attributes transmitted to SAML and OAuth applications for a specific user.

-   **Display applications a user has access to**

>     An administrator can now see all applications a user has access to on the user's page in the admin interface.

Reviewer: ```suggestion
    An administrator can now see all applications a user has access to on the user's page in the Admin interface.
```

### Other noteworthy features

-   **New graph for event volume**

>     In the **Events** -> **Log** page in the admin interface, you can see a graph of the event volume matching the search query over the last 7 days.

Reviewer: ```suggestion
    In the **Events** -> **Log** page in the Admin interface, you can see a graph of the event volume matching the search query over the last 7 days.
```

-   **Flows can now be restricted to outposts**

>     You can now restrict a flow to be used only by an outpost using the **Require Outpost** setting on the flow. This would mainly be used to LDAP flows.

Reviewer: ```suggestion
    You can now restrict a flow to be used only by an outpost using the **Require Outpost** setting on the flow. This would mainly be used for LDAP flows.
```

---
title: Release 2024.2
slug: /releases/2024.2
---
:::::note

> 2024.2 has not been released yet! We're publishing these release notes as a preview of what's to come, and for our awesome beta testers trying out release candidates.

Author: Should we add an `(unreleased)` in the page title?

-   User sessions will be invalidated after this upgrade. As such, users will need to log back in.

-   The Helm Chart has a number of breaking changes. Find out more in the [chart release notes](https://github.com/goauthentik/helm/releases/tag/authentik-2024.2.0).

> ## New features

Author: Missing the new application/provider creation wizard I think?
